tests/foreman/cli/test_product.py

```diff
@@ -140,6 +140,7 @@ def test_positive_CRUD(self):
         })
         self.assertEqual(len(product['sync-plan-id']), 0)
         Product.delete({u'id': product['id']})
+        time.sleep(5)
```

Comment:
Removed my last comment as not applicable for CLI ... however would like to know why do we specifically want to wait for `5` secs ?
Can we give a try to  use `wait_for` so that on each run test would wait every time for `5 ` sec ?

Reply:
Please refer i.e.  and could use `delay=1` for more frequent check .. https://github.com/SatelliteQE/robottelo/blob/099dcf1d947d95f721083bbb7aac566a6b0fc3f0/tests/foreman/cli/test_repository.py#L133-L139

Reply:
Or another way is you could use below to wait for complete the task, i.e. 

https://github.com/SatelliteQE/robottelo/blob/2599f5d763746f0553759291f315e759a85541e5/tests/foreman/longrun/test_inc_updates.py#L199-L204